librz/core/cmd/cmd_type.c

```diff
@@ -798,3 +798,76 @@ RZ_IPI RzCmdStatus rz_type_xrefs_list_all_handler(RzCore *core, int argc, const
 	types_xrefs_all(core);
 	return RZ_CMD_STATUS_OK;
 }
+
+RZ_IPI RzCmdStatus rz_type_class_show_handler(RzCore *core, int argc, const char **argv) {
+	RzBaseType *btype = rz_type_db_get_base_type(core->analysis->typedb, argv[1]);
+	if (!btype) {
+		RZ_LOG_ERROR("Cannot find \"%s\" type\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (btype->kind != RZ_BASE_TYPE_KIND_ATOMIC && btype->kind != RZ_BASE_TYPE_KIND_TYPEDEF) {
+		RZ_LOG_ERROR("\"%s\" type is not atomic or typedef\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzTypeTypeclass tclass = rz_base_type_typeclass(core->analysis->typedb, btype);
+	rz_cons_println(rz_type_typeclass_as_string(tclass));
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_type_class_set_handler(RzCore *core, int argc, const char **argv) {
+	RzBaseType *btype = rz_type_db_get_base_type(core->analysis->typedb, argv[1]);
+	if (!btype) {
+		RZ_LOG_ERROR("Cannot find \"%s\" type\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (btype->kind != RZ_BASE_TYPE_KIND_ATOMIC && btype->kind != RZ_BASE_TYPE_KIND_TYPEDEF) {
+		RZ_LOG_ERROR("\"%s\" type is not atomic\n", argv[1]);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzTypeTypeclass tclass = rz_type_typeclass_from_string(argv[2]);
+	rz_cons_println(rz_type_typeclass_as_string(tclass));
+	return RZ_CMD_STATUS_OK;
```

Comment:
Where is the typeclass *set* here?

Reply:
It's a draft pull request, no need to review just yet. I am working on "set" since it can be complicated because it need to propagate all over the chain of typedefs, etc